### PR TITLE
Enable field ionization from PICMI

### PR DESCRIPTION
**Pull request by NeilZaim**
The goal of this PR is to take into account in WarpX the changes made in this PICMI Pull Request: https://github.com/picmi-standard/picmi/pull/75.

I haven't tested it yet because I first wanted to make sure that this is generally how it should be done. cc @dpgrote

Also, how do we synchronously merge this PR and picmi's PR? Is the general workflow to
1. Merge the picmi PR
2. Update the picmi version
3. Use the new version inside this PR and then we can merge it? (and we possibly have to do this for other codes that use picmi like warp or fbpic)

### CODE REVIEW

File: Python/pywarpx/picmi.py [hunk 1]
```diff
@@ -163,14 +163,13 @@ def initialize_inputs(self, layout,
...
 
-        for interaction in self.interactions:
-            assert interaction[0] == 'ionization'
-            assert interaction[1] == 'ADK', 'WarpX only has ADK ionization model implemented'
-            self.species.do_field_ionization=1
-            self.species.physical_element=self.particle_type
-            self.species.ionization_product_species = interaction[2].name
-            self.species.ionization_initial_level = self.charge_state
-            self.species.charge = 'q_e'
+    def activate_ionization(self, product_species):
```
Comment (dpgrote) [inline Python/pywarpx/picmi.py]: Thanks for the PR. This looks ok, but is different than what I would have done. I would have put the code here in the  `initialize_inputs` method in the `FieldIonization` class. This more closely follows the pattern of how things are done in PICMI. That class has the references to the species and so can set these parameters that are related to the field ionization setup. Also, there is no need to use `add_new_attr` - these parameters can be set directly, `self.species.do_field_ionization = 1`. `add_new_attr` is there for when the attributes have special characters or is constructed.

File: Python/pywarpx/picmi.py [hunk 2]
```diff
@@ -1126,7 +1119,11 @@ def initialize_inputs(self):
 class FieldIonization(picmistandard.PICMI_FieldIonization):
     def initialize_inputs(self):
         assert self.model == 'ADK', 'WarpX only has ADK ionization model implemented'
```
Comment (dpgrote) [inline Python/pywarpx/picmi.py:1216]: Can you add the WarpX specific doc string for this class that states that only ADK is supported here?

Comment (NeilZaim) [reply Python/pywarpx/picmi.py:1216]: I've just added a docstring which I hope is what is needed here.

**Comment by dpgrote** [comment]
I'll update the picmi version now that PR #75 was merged. picmistandard is now up to version 0.0.21.

**Comment by NeilZaim** [comment]
@dpgrote Sorry I had abandonned this PR for a while, but it should be ready now.

**Review by dpgrote — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**